doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -756,15 +756,77 @@ Subrelations are implemented in ``Classes.Morphisms`` and are a prime
 example of a mostly user-space extension of the algorithm.
 
 
-Constant unfolding
-~~~~~~~~~~~~~~~~~~
-
-The resolution tactic is based on typeclasses and hence regards user-defined
-:term:`constants <constant>` as transparent by default. This may slow down the
-resolution due to a lot of unifications (all the declared ``Proper``
-instances are tried at each node of the search tree). To speed it up,
-declare your constant as rigid for proof search using the command
-:cmd:`Typeclasses Opaque`.
+Constant unfolding during rewriting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, :tacn:`setoid_rewrite` and :tacn:`rewrite_strat` unfold global
+definitions during rewrite rule matching, but do not unfold local
+definitions. Unfolding definitions may slow down matching, whereas keeping
+definitions opaque may cause matches to be missed.  This behavior is
+configurable using transparency hints in the hint database ``rewrite``:
```

Comment:
FWIW: Do you think the variations in how transparent and opaque work for various tactics makes sense or just sort of happened?  If we think it's the latter, making the behavior more consistent may break current projects, so probably not something we will take on :-(

Reply:
I'm not sure. The process I imagine might have led to the current state of affairs is one of trying to provide some configurability without designing and implementing completely new mechanisms. There clearly is some understanding that this ought to be improved on, as the Ltac2 reduction tactics already take options as first-class arguments. As for what to do with existing code, I am cautiously optimistic: for example, a substantial part of stdlib and CI builds with `Hint Constants Opaque : rewrite` (which I wonder might be a less surprising default).

Reply:
`Hint Constants Opaque : rewrite / typeclass_instances` ought to be there in a GoodDefaults.v file certainly @thomas-lamiaux

Reply:
I suggest we work on it at next CUDW if we haven't by then